### ISSUE TITLE
Automount hangs file operations when device is not present. Ignores timeout.

### ISSUE THREAD

**Issue by beanaroo** [systemd/systemd#2181]
I have external drives that are automounted on demand and unmounted on idle nicely. Problem is, when a drive is not present, operations (like when the above directory is accessed) hang for 90sec. The `x-systemd.device-timeout` fstab option works nicely for `nofail` cases but seems to be of no use for `systemd.automount` cases.

---

I have noted, according to [systemd.mount(5)](http://www.freedesktop.org/software/systemd/man/systemd.mount.html#x-systemd.device-timeout=), that device-timeout is only valid in fstab and not `Options=`. 

If I understand correctly, this is why the fstab entry:

```
UUID=12345  /home/user/Media/Backup     ext4    noauto,x-systemd.automount,x-systemd.device-timeout=1,x-systemd.idle-timeout=10min     0 0 
```

...is not reflected in `systemctl show -a home-user-Media-Backup.mount` (generated):

```
Where=/home/user/Media/Backup
What=/dev/disk/by-uuid/12345
Options=noauto,x-systemd.automount,x-systemd.idle-timeout=10min
Type=ext4
TimeoutUSec=1min 30s
```

---

Is this by design or a limitation or just unimplemented? I have come across [others](http://forums.debian.net/viewtopic.php?f=5&t=123704) with the [same problem](https://bbs.archlinux.org/viewtopic.php?id=186030).

**Comment by arvidjaar** [comment]
`x-systemd.device-timeout` should be translated to `JobTimeoutSec`; do you have it defined?

**Comment by beanaroo** [comment]
`JobTimeoutUSec` = `0`, so undefined?

**Comment by poettering** [comment]
its more complicated... the device timeout is actually set on the .device unit, not the mount. Hence, to achieve the same effect from unit files, you'd have to add a .device drop-in, not make a .mount change...

**Comment by poettering** [comment]
Fix waiting in #3170.
